src/test/java/com/thebois/models/beings/ColonyTests.java

```diff
@@ -49,4 +61,78 @@ public void updateCallsUpdateOnAllBeings() {
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("getItemTypes")
+    public void emptyColonyInventoryIsEmpty(final ItemType itemType) {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        final Optional<IItem> result = colony.takeItem(itemType);
+
+        // Assert
+        assertThat(result.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void canAddAndTakeItemToInventory() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        colony.addItem(new Log());
+        final Optional<IItem> item = colony.takeItem(ItemType.LOG);
+
+        // Assert
+        assertThat(item.isPresent()).isTrue();
+        assertThat(item.get().getType()).isEqualTo(ItemType.LOG);
+    }
+
+    @Test
+    public void countIsZeroWhenInventoryIsEmpty() {
+        // Arrange
+        final Colony colony = mockColony();
+
+        // Act
+        final int count = colony.getItemCount(ItemType.ROCK);
+
+        // Assert
+        assertThat(count).isEqualTo(0);
+    }
+
+    @Test
+    public void countIsNotZeroWhenInventoryGotSpecifiedItemTypeInIt() {
```

Comment:
style

I'm re-adding this comment because the name of the test still doesn't match what the test actually asserts.

Either change "IsNotZero" in the name to "IsCorrectCount", or change `assertThat(count).isEqualTo(2)` to `assertThat(count).isNotEqualTo(0)`

Reply:
👍